### PR TITLE
Changes after discussions

### PR DESCRIPTION
**Pull request by andrewsparkes**

### CODE REVIEW

File: tables_script.sql [hunk 1]
```diff
@@ -1,28 +1,29 @@
 -- Automation system runs table
 CREATE TABLE IF NOT EXISTS `biosero_uat`.`automation_system_runs` (
   `id` INT NOT NULL AUTO_INCREMENT,
-  `automation_system` VARCHAR(255) NOT NULL,
-  `run_id` INT NOT NULL UNIQUE,
+  `automation_system_type` ENUM('biosero', 'beckman', 'sentinel') NOT NULL,
```
Comment (pjvv) [inline tables_script.sql]: I would drop 'beckman' and 'sentinel' while we do not support them - to keep things simple and concise.

File: tables_script.sql [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+  `automation_system_name` ENUM('CPA', 'CPB') NOT NULL,
+  `system_run_id` INT NOT NULL UNIQUE,
```
Comment (pjvv) [inline tables_script.sql]: Can we add comments to a create query? Might be worth adding some to know the differences between the `id` and `system_run_id`; especially for those who haven't been in the discussions and for me in one months time when I've forgotten everything!

File: tables_script.sql [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
   `method` VARCHAR(255) NOT NULL,
   `user_id` VARCHAR(255) NOT NULL,
   `start_time` DATETIME DEFAULT CURRENT_TIMESTAMP NOT NULL,
   `end_time` DATETIME DEFAULT NULL,
-  `state` VARCHAR(255) NOT NULL,
-  `machine_name` VARCHAR(255) NOT NULL,
-  `instrument_serial_number` VARCHAR(255) NOT NULL,
+  `state` ENUM('started','completed','aborted') NOT NULL,
+  `liquid_handler_serial_number` ENUM('h1000001','h1000002') NOT NULL,
```
Comment (pjvv) [inline tables_script.sql]: 👍🏼

File: tables_script.sql [hunk 4]
```diff
@@ -63,14 +64,21 @@ CREATE TABLE IF NOT EXISTS `biosero_uat`.`destination_plate_wells` (
...
-  `coordinate` VARCHAR(255) NOT NULL,
+  `coordinate` ENUM('A1','A2','A3','A4','A5','A6','A7','A8','A9','A10','A11','A12','B1','B2','B3','B4','B5','B6','B7','B8','B9','B10','B11','B12','C1','C2','C3','C4','C5','C6','C7','C8','C9','C10','C11','C12','D1','D2','D3','D4','D5','D6','D7','D8','D9','D10','D11','D12','E1','E2','E3','E4','E5','E6','E7','E8','E9','E10','E11','E12','F1','F2','F3','F4','F5','F6','F7','F8','F9','F10','F11','F12','G1','G2','G3','G4','G5','G6','G7','G8','G9','G10','G11','G12','H1','H2','H3','H4','H5','H6','H7','H8','H9','H10','H11','H12') NOT NULL,
   `source_plate_well_id` INT DEFAULT NULL,
   `control_plate_well_id` INT DEFAULT NULL,
   `created_at` DATETIME DEFAULT CURRENT_TIMESTAMP NOT NULL,
   `updated_at` DATETIME DEFAULT CURRENT_TIMESTAMP NOT NULL,
   PRIMARY KEY (`id`),
   KEY `index_destination_plate_wells_on_barcode` (`barcode`),
   CONSTRAINT `run_destination_plate_well` UNIQUE (`automation_system_run_id`,`barcode`,`coordinate`),
+  CONSTRAINT `source_plate_well_id_or_control_plate_well_id` CHECK (
```
Comment (pjvv) [inline tables_script.sql:75]: @harrietc52, I haven't tried it myself but would [this](https://dev.mysql.com/doc/refman/8.0/en/logical-operators.html#operator_xor) work?

File: README.md [hunk 5]
```diff
@@ -41,217 +41,1885 @@ The required view creation scripts are found in:
...
 ```sql
 SELECT
-    id, machine_name, name, value, description, created_at
+    id, automation_system_name, config_key, config_value, description, created_at
 FROM
-    `biosero_test`.`configurations`
+    `biosero_uat`.`configurations`
 WHERE
-    machine_name = 'Workcell 1'
+    machine_name = 'CPA'
```
Comment (harrietc52) [inline README.md:48]: What does CPA stand for?

File: tables_script.sql [hunk 6]
```diff
@@ -50,7 +51,7 @@ CREATE TABLE IF NOT EXISTS `biosero_uat`.`control_plate_wells` (
   `id` INT NOT NULL AUTO_INCREMENT,
   `automation_system_run_id` INT NOT NULL,
   `barcode` VARCHAR(255) NOT NULL,
-  `coordinate` VARCHAR(255) NOT NULL,
+  `coordinate` ENUM('A1','A2','A3','A4','A5','A6','A7','A8','A9','A10','A11','A12','B1','B2','B3','B4','B5','B6','B7','B8','B9','B10','B11','B12','C1','C2','C3','C4','C5','C6','C7','C8','C9','C10','C11','C12','D1','D2','D3','D4','D5','D6','D7','D8','D9','D10','D11','D12','E1','E2','E3','E4','E5','E6','E7','E8','E9','E10','E11','E12','F1','F2','F3','F4','F5','F6','F7','F8','F9','F10','F11','F12','G1','G2','G3','G4','G5','G6','G7','G8','G9','G10','G11','G12','H1','H2','H3','H4','H5','H6','H7','H8','H9','H10','H11','H12') NOT NULL,
   `control` VARCHAR(20) DEFAULT NULL,
```
Comment (pjvv) [inline tables_script.sql]: This could be a ENUM too?

File: tables_script.sql [hunk 7]
```diff
@@ -82,13 +90,13 @@ CREATE TABLE IF NOT EXISTS `biosero_uat`.`destination_plate_wells` (
 -- Configurations table
 CREATE TABLE IF NOT EXISTS `biosero_uat`.`configurations` (
   `id` INT NOT NULL AUTO_INCREMENT,
-  `machine_name` VARCHAR(255) NOT NULL,
+  `automation_system_name` VARCHAR(255) NOT NULL,
```
Comment (pjvv) [inline tables_script.sql]:
```suggestion
  `automation_system_name` ENUM('CPA', 'CPB') NOT NULL,
```

File: README.md [hunk 8]
```diff
@@ -41,217 +41,1885 @@ The required view creation scripts are found in:
...
 ORDER BY id ASC;
 ```
 
-Example JSON representation for workcell 1 configuration:
+Example JSON representation for CPA configuration:
```
Comment (harrietc52) [inline README.md]: What is this referring to? The different config_key 's ?

File: README.md [hunk 9]
```diff
@@ -41,217 +41,1885 @@ The required view creation scripts are found in:
...
+INSERT INTO `biosero_uat`.`control_plate_wells` (
+  automation_system_run_id,
+  barcode,
+  coordinate,
+  control,
+  created_at,
+  updated_at
+)
+VALUES
+(@a_system_run_id, 'Control01', 'A1', 'positive', now(), now())
```
Comment (harrietc52) [inline README.md:284]: Is it worth making the control fiend case insensitive

File: tables_script.sql (hunk 1 above)
Comment (andrewsparkes) [reply tables_script.sql]: changed

File: tables_script.sql (hunk 6 above)
Comment (andrewsparkes) [reply tables_script.sql]: control? yes, changed

File: tables_script.sql (hunk 7 above)
Comment (andrewsparkes) [reply tables_script.sql]: done

File: tables_script.sql (hunk 6 above)
Comment (harrietc52) [inline tables_script.sql]: Could `control` also be an ENUM, and can it be NULL?

File: tables_script.sql (hunk 2 above)
Comment (andrewsparkes) [reply tables_script.sql]: yes, adding

File: view_run_level_view.sql [hunk 10]
```diff
@@ -1,22 +1,23 @@
 -- Run Level View
 CREATE VIEW `biosero_uat`.`run_level_view` AS
     SELECT
-        r.automation_system,
-        r.run_id,
+        r.automation_system_type,
+        r.automation_system_name,
+        r.system_run_id,
         r.method,
         r.start_time,
```
Comment (harrietc52) [inline view_run_level_view.sql:9]: user?

**Review by harrietc52 — APPROVED** [review]
Minor thoughts 👍

File: tables_script.sql (hunk 4 above)
Comment (harrietc52) [reply tables_script.sql:75]: Yep XOR, I'll have a look into it @pjvv

Comment (harrietc52) [reply tables_script.sql:75]: Could replace with
```sql
  CONSTRAINT `source_plate_well_id_or_control_plate_well_id` CHECK (
	`control_plate_well_id` = 1 XOR `source_plate_well_id` = 1   
  )
```

Comment (andrewsparkes) [reply tables_script.sql:75]: changed

File: README.md (hunk 5 above)
Comment (andrewsparkes) [reply README.md:48]: Cherrypick workcell A
As opposed to the Beckmans which are called CP1 to 4
A bit limited but RnD decided these names

File: README.md (hunk 9 above)
Comment (andrewsparkes) [reply README.md:284]: Made it an enum with positive and negative values only

File: tables_script.sql (hunk 6 above)
Comment (andrewsparkes) [reply tables_script.sql]: yes changed it

File: view_run_level_view.sql (hunk 10 above)
Comment (andrewsparkes) [reply view_run_level_view.sql:9]: added

File: README.md (hunk 8 above)
Comment (andrewsparkes) [reply README.md]: The set of config keys for workcell 'CPA', will make clearer

File: tables_script.sql [hunk 11]
```diff
@@ -1,83 +1,79 @@
 -- Automation system runs table
 CREATE TABLE IF NOT EXISTS `biosero_uat`.`automation_system_runs` (
```
Comment (harrietc52) [inline tables_script.sql:2]: Could also add
```sql
CREATE DATABASE IF NOT EXISTS `biosero_uat`;
```

**Pull request merged**